awstesting/integration/performance/s3DownloadManager/main_test.go

```diff
@@ -0,0 +1,115 @@
+// +build integration,perftest
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/awstesting/integration"
+	"github.com/aws/aws-sdk-go/internal/sdkconstants"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+var benchConfig BenchmarkConfig
+
+type BenchmarkConfig struct {
+	bucket       string
+	tempdir      string
+	clientConfig ClientConfig
+}
+
+func (b *BenchmarkConfig) SetupFlags(prefix string, flagSet *flag.FlagSet) {
+	flagSet.StringVar(&b.bucket, "bucket", "", "Bucket to use for benchmark")
+	flagSet.StringVar(&b.tempdir, "temp", os.TempDir(), "location to create temporary files")
+	b.clientConfig.SetupFlags(prefix, flagSet)
+}
+
+var benchStrategies = []struct {
+	name           string
+	bufferProvider s3manager.WriterReadFromProvider
+}{
+	{name: "Unbuffered", bufferProvider: nil},
+	{name: "Buffered", bufferProvider: s3manager.NewPooledBufferedWriterReadFromProvider(int(sdkconstants.MebiByte))},
```

Comment:
Minor style suggestion. Could use a `map[string]struct` instead of a struct with a `name` member. Will accomplish the same goal, just organizes the test cases so `name` is more obvious/upfront..